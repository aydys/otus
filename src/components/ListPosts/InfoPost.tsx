import React from 'react'

type Props = {
  posts: any[],
  post: {}
}

export class InfoPost extends React.Component<Props> {
  render() {
    return <div>InfoPost</div>
  }
}