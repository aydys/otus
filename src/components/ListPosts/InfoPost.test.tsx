import React from "react";
import { shallow } from "enzyme";
import { InfoPost } from "./InfoPost";

describe('ListPosts', () => {
  
  const props = {
    post: {
      title: '',
      id: 1,
      body: ''
    }
  }

  const wrapper = shallow(<InfoPost {...props} />)
  it('layout dom elements (snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  describe('InfoPost with props', () => {
    const nextProps = {
      ...props,
      post: {
        ...props.post,
        title: 'Some title'
      }
    }

    const wrapper = shallow(<InfoPost {...nextProps} />)

    it('render Title of Post', () => {
      expect(wrapper.find('h4').text()).toEqual(nextProps.post.title);
    })
  })  
})
