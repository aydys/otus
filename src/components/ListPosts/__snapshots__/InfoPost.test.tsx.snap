// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListPosts layout dom elements (snapshot 1`] = `
<Fragment>
  <h4 />
  <p />
</Fragment>
`;
